src/charm.py

```diff
@@ -83,14 +88,41 @@ def _on_collect_unit_status(self, event: CollectStatusEvent):
             return
         event.add_status(ActiveStatus())
 
+    def _is_ca_cert_active(self) -> bool:
+        """Return whether the CA certificate is active by checking the secret expiry."""
+        secret = self.model.get_secret(label=CA_CERTIFICATES_SECRET_LABEL)
+        secret_info = secret.get_info()
+        if not secret_info.expires:
+            return False
+        if secret_info.expires.tzinfo is None:
+            return secret_info.expires > datetime.now()
+        return secret_info.expires > datetime.now(secret_info.expires.tzinfo)
+
     @property
-    def _config_root_ca_certificate_validity(self) -> int:
-        """Return Root CA certificate validity (in days).
+    def _config_root_ca_certificate_validity(self) -> timedelta:
+        """Return Root CA certificate validity (timedelta)."""
+        return self._int_config_to_timedelta(
+            int(self.model.config.get("root-ca-validity")),  # type: ignore[arg-type]
+            self._config_validity_unit,
+        )
+
+    @property
+    def _config_validity_unit(self) -> str:
+        """Return Validity unit.
 
         Returns:
-            int: Certificate validity (in days)
+            str: Validity unit.
         """
-        return int(self.model.config.get("root-ca-validity"))  # type: ignore[arg-type]
+        return str(self.model.config.get("validity-unit", "days"))  # type: ignore[arg-type]
+
+    @property
+    def _ca_certificate_renewal_threshold(self) -> timedelta:
+        """Return CA certificate renewal threshold.
+
+        Which is the time difference between the validity of the root certificate
+        and issued certificates.
+        """
```

Comment:
I know the math is quite trivial, but an example would be useful here.

Reply:
✅